cpp/BUILDING.md

```diff
@@ -0,0 +1,20 @@
+# C++ Binding instructions
+
+The building of the CPP library is currently not done by CI, so if you're wanting to produce C/C++ bindings and the associated
+library you'll need to follow the following steps.
+
+## Prerequisites
+
+- C++ 17 or greater installed. Most modern gcc or clang compilers should do
+- C++ Boost installed. Search up directions for your distribution if you're not sure.
```

Comment:
Do we know what pieces of boost we require? On unbuntu installed a meta-package for `libboost-all-dev`, which installed 100+ packages and took 500 MB of space. :-| If we don't know which pieces we need, we should write a ticket to find out.

Reply:
We only need optional, but I don't know what that requires.